samples/go/client/client.go

```diff
@@ -0,0 +1,146 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"a2a/models"
+)
+
+// Client represents an A2A protocol client
+type Client struct {
+	baseURL    string
+	httpClient *http.Client
+}
+
+// NewClient creates a new A2A client
+func NewClient(baseURL string) *Client {
```

Comment:
This is more of an enhancement, and not a requirement, but it would be nice to have a `NewClientFromAgentCard` or something of the like, where you have an `AgentCard` created/pulled from a remote, and use the URL from within it.

Again, not a blocker but something to consider.

Reply:
By the way, I don't care as much about naming so much as I do the ability to create a new A2A client/server manually _(no `AgentCard`)_ and via the `AgentCard`.

Reply:
Done